packages/blockchain-wallet-v4-frontend/src/data/components/sendBsv/sagas.js

```diff
@@ -0,0 +1,246 @@
+import { call, select, put } from 'redux-saga/effects'
+import { equals, path, prop, nth, is, identity } from 'ramda'
+import * as A from './actions'
+import * as S from './selectors'
+import { FORM } from './model'
+import { actions, model, selectors } from 'data'
+import settings from 'config'
+import {
+  initialize,
+  change,
+  startSubmit,
+  stopSubmit,
+  destroy
+} from 'redux-form'
+import * as C from 'services/AlertService'
+import { promptForSecondPassword } from 'services/SagaService'
+import { Exchange, Remote } from 'blockchain-wallet-v4/src'
+import { ADDRESS_TYPES } from 'blockchain-wallet-v4/src/redux/payment/btc/utils'
+
+export const logLocation = 'components/sendBsv/sagas'
+export const bsvDefaultFee = 4
+
+export default ({ coreSagas }) => {
+  const initialized = function*() {
+    try {
+      yield put(A.sendBsvPaymentUpdated(Remote.Loading))
+      let payment = coreSagas.payment.bsv.create({
+        network: settings.NETWORK_BSV
+      })
+      payment = yield payment.init()
+      const accountsR = yield select(
+        selectors.core.common.bsv.getAccountsBalances
+      )
+      const defaultIndexR = yield select(
+        selectors.core.kvStore.bsv.getDefaultAccountIndex
+      )
+      const defaultIndex = defaultIndexR.getOrElse(0)
+      const defaultAccountR = accountsR.map(nth(defaultIndex))
+      payment = yield payment.from(defaultIndex, ADDRESS_TYPES.ACCOUNT)
+      payment = yield payment.fee(bsvDefaultFee)
+      const initialValues = {
+        coin: 'BSV',
+        from: defaultAccountR.getOrElse()
+      }
+      yield put(initialize(FORM, initialValues))
+      yield put(A.sendBsvPaymentUpdated(Remote.of(payment.value())))
+    } catch (e) {
+      yield put(
+        actions.logs.logErrorMessage(logLocation, 'sendBsvInitialized', e)
+      )
+    }
+  }
+
+  const destroyed = function*() {
+    yield put(actions.form.destroy(FORM))
+  }
+
+  const firstStepSubmitClicked = function*() {
+    try {
+      let p = yield select(S.getPayment)
+      yield put(A.sendBsvPaymentUpdated(Remote.Loading))
+      let payment = coreSagas.payment.bsv.create({
+        payment: p.getOrElse({}),
+        network: settings.NETWORK_BSV
+      })
+      payment = yield payment.build()
+      yield put(A.sendBsvPaymentUpdated(Remote.of(payment.value())))
+    } catch (e) {
+      yield put(
+        actions.logs.logErrorMessage(logLocation, 'firstStepSubmitClicked', e)
+      )
+    }
+  }
+
+  const formChanged = function*(action) {
+    try {
+      const form = path(['meta', 'form'], action)
+      const field = path(['meta', 'field'], action)
+      const payload = prop('payload', action)
+      if (!equals(FORM, form)) return
+
+      let p = yield select(S.getPayment)
+      let payment = coreSagas.payment.bsv.create({
+        payment: p.getOrElse({}),
+        network: settings.NETWORK_BSV
+      })
+
+      switch (field) {
+        case 'coin':
+          switch (payload) {
```

Comment:
BCH is missing from the switch

Reply:
None of these are needed actually since send bsv does not let you select diff coin. Will delete